lib/core/versions/latest/CreateOperation.ts

```diff
@@ -186,7 +186,7 @@ export default class CreateOperation implements OperationModel {
       throw new SidetreeError(ErrorCode.CreateOperationMissingOrUnknownProperty);
     }
 
-    const encodedSuffixData = operationObject.suffix_data;
```

Comment:
Legacy long form has to remain snake case.

Reply:
legacy clients should pin their version to avoid taking this change... semver allows them to avoid taking a breaking change until they are ready to.

Reply:
any clients that are using the legacy version, should pin to the last stable version until they are ready to take breaking changes.

semver allows client to take breaking changes when they would like to, and avoid taking them until they are ready to "catch up".

Reply:
That's one valid view of how a node operator can support (or not support) breaking long-form DID changes before v1, this approach would require 2 versions of Sidetree nodes to be deployed side-by-side. But the reference implementation took a different approach, ie. single version deployment that supports such breaking changes, not supporting this now would mean a lot more work from Microsoft's DID infra team and some work on clients/services teams to understand 2 versions and 2 endpoints, which would almost certainly mean push back on the casing change. The idea always was to minimize the downstream dependent client/service changes.

Reply:
I don't understand your comment, how does pinning the version not solve this problem? 

ION is already pinned:

https://github.com/decentralized-identity/ion/blob/master/package.json#L9

The latest unstable version is https://www.npmjs.com/package/@decentralized-identity/sidetree/v/0.11.0-unstable.8c59559

This change set will break the ability for clients to resolve long form DIDs, of the (now 2) legacy formats.... You should simply avoid taking these changes until you are willing to support them.

Reply:
@OR13 I think the concern is that this means they have to keep separate nodes running for multiple teams until they're all up to date.

@thehenrytsai Is this legacy code that's just going to be deprecated in the future and it's fine to make this change where it's completely removed in a later PR once the client teams can update to not using this longform anymore?

Reply:
my concern is that a "reference implementation" should not be used to support "legacy code paths".... it is meant to be copied by other implementations... I suggest pinning versions, and removing any "legacy code" from the sidetree reference implementation.

Reply:
From what I'm understanding, they want it removed, and they want these new changes to start building off ASAP, but would end up blocking other internal teams and causing lots of unnecessary, regretful work if they have to accept both at once.

It seems reasonable to me to leave them in for a bit longer until they're able to get around to updating them especially since we didn't set an official deprecation date originally from my understanding. This has been what we've done in the Aries/Indy-SDK world which has worked very well for us. As an example, (the precursor authcrypt function)[https://github.com/hyperledger/indy-sdk/blob/f9eb2cf17b51584f875c4707094256a96656e7b8/libindy/src/api/crypto.rs#L288] still exists in Indy-sdk even though everyone is well beyond that work. It's not completely unreasonable to wait a bit longer on this change until things can get updated even with version pinning. Would you be alright if we left it in for now with a comment saying "these are legacy code paths and MUST NOT be implemented by reference implementors going forward"?

Reply:
@kdenhartog, your understanding is accurate, we intend to remove legacy behavior at the first possible opportunity, certainly prior to v1, we just wanted to give whoever is dependent on current behavior (including internal MS teams) the some time to switch over instead of a full breakage. I agree in a perfect world I also would rather not have this stop-gap.